From f2c4db4321c2ee3a4828dae789c0ec3ab71bafa7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Petr=20P=C3=ADsa=C5=99?= <ppisar@redhat.com>
Date: Wed, 19 Mar 2014 11:03:08 +0100
Subject: [PATCH] Add support for the basic operations necessary to support
 ECDH
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This patch adds EC_KEY_new_by_curve_name(), EC_KEY_free*() and
SSL_CTX_set_tmp_ecdh().

This is ported from Net-SSLeay-1.56 to version 1.35.

Signed-off-by: Petr Písař <ppisar@redhat.com>
---
 SSLeay.xs | 17 +++++++++++++++++
 typemap   |  1 +
 2 files changed, 18 insertions(+)

diff --git a/SSLeay.xs b/SSLeay.xs
index 558da0b..5bec132 100644
--- a/SSLeay.xs
+++ b/SSLeay.xs
@@ -2456,6 +2456,23 @@ SSL_CTX_set_tmp_rsa(ctx,rsa)
      SSL_CTX *	ctx
      RSA *	rsa
 
+#if OPENSSL_VERSION_NUMBER > 0x10000000L
+
+EC_KEY *
+EC_KEY_new_by_curve_name(nid)
+    int nid
+
+void
+EC_KEY_free(key)
+    EC_KEY * key
+
+long
+SSL_CTX_set_tmp_ecdh(ctx,ecdh);
+     SSL_CTX * ctx
+     EC_KEY  * ecdh
+
+#endif
+
 void *
 SSL_get_app_data(s)
      SSL *	s
diff --git a/typemap b/typemap
index b97be80..698790b 100644
--- a/typemap
+++ b/typemap
@@ -5,6 +5,7 @@ SSL_SESSION *	T_IV
 SSL *		T_IV
 RSA *		T_IV
 DH *		T_IV
+EC_KEY *	T_IV
 X509 *		T_IV
 X509_CRL *	T_IV
 X509_LOOKUP *	T_IV
-- 
1.8.5.3

